### ISSUE TITLE
bug: Very slow first visit

### ISSUE THREAD

**Issue by aoozoo** [apache/apisix#5559]
### Issue description

# Very slow first visit

![截图](https://user-images.githubusercontent.com/15232738/142590356-26750c62-ab9a-42cf-a03b-6014dc02eb83.png)

### Environment

- apisix version: 2.10.1
- OS: Centos7
- OpenResty / Nginx version: 1.19.9.1
- etcd version, if have (cmd: run `` to get the info from server-info API): 3.5.0

### Steps to reproduce

1. Create a routing rule and forward it to a back-end service
2. Access through the browser, match the route for processing
3. Open F12 and you will find that the first request takes more than 4 seconds to return

### Actual result

3. Open F12 and you will find that the first request takes more than 4 seconds to return

### Error log

No error

### Expected result

Why does the first request take more than 4 seconds, and it will be returned after a few tens of milliseconds after refreshing. The back-end service is a dynamic interface, and the interface is processed in 1 millisecond.
The first request should be returned in tens of milliseconds.

**Comment by shuaijinchao** [comment]
Is there any abnormal information in the error.log?

**Comment by tokers** [comment]
Can you reproduce it by `curl`.

**Comment by moonming** [comment]
@aoozoo 
Please use English in the public channel, thx

**Comment by aoozoo** [comment]
# route config
```json
{
  "uri": "/testslow",
  "name": "testslow",
  "methods": [
    "GET",
    "POST",
    "PUT",
    "DELETE",
    "PATCH",
    "HEAD",
    "OPTIONS",
    "CONNECT",
    "TRACE"
  ],
  "plugins": {
    "proxy-rewrite": {
      "regex_uri": [
        "/(.*)",
        "/get?foo1=bar1&foo2=$1"
      ]
    }
  },
  "upstream": {
    "nodes": [
      {
        "host": "httpbin.org",
        "port": 80,
        "weight": 1
      }
    ],
    "timeout": {
      "connect": 6,
      "send": 6,
      "read": 6
    },
    "type": "roundrobin",
    "scheme": "http",
    "pass_host": "node",
    "keepalive_pool": {
      "idle_timeout": 60,
      "requests": 1000,
      "size": 320
    }
  },
  "status": 1
}
```


# tcpdump when use firefox
![image](https://user-images.githubusercontent.com/15232738/142602469-85efb3d0-1476-4369-ae03-e8ac76067f16.png)


# tcpdump when use curl
![image](https://user-images.githubusercontent.com/15232738/142602759-8f75d5be-a1f0-4a39-876d-5d38cf2c31cf.png)

![image](https://user-images.githubusercontent.com/15232738/142602860-26f366a3-fd62-4c35-89f2-a438ca168ec3.png)


# It looks like a connection problem

**Comment by aoozoo** [comment]
# tail error logs
```bash
[root@Node_10.10.10.21 ~]#  tail /usr/local/apisix/logs/error.log 
2021/11/19 16:10:34 [warn] 7797#7797: *7 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7802#7802: *6 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7799#7799: *2 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7800#7800: *4 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7803#7803: *9 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7798#7798: *1 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7801#7801: *5 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7806#7806: *8 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 16:10:34 [warn] 7796#7796: *3 [lua] plugin.lua:192: load_stream(): new plugins: {"ip-restriction":true,"limit-conn":true,"mqtt-proxy":true}, context: init_worker_by_lua*
2021/11/19 17:32:06 [error] 7800#7800: *567276 upstream prematurely closed connection while reading response header from upstream, client: 10.10.10.1, server: _, request: "GET /testslow HTTP/1.1", upstream: "http://54.156.165.4:80/get?foo1=bar1&foo2=testslow", host: "10.10.10.21:9080"
[root@Node_10.10.10.21 ~]#  
```

**Comment by aoozoo** [comment]
## I think you can reproduce this problem

**Comment by aoozoo** [comment]
tcpdum from apisix server with ip 10.10.10.21

**Comment by juzhiyuan** [comment]
@aoozoo Hi, this issue won't happen due to the apisix itself usually, but we still need to have a check on what happened here, to make sure apisix is ok.

Back to your issue, I noticed there has a `keepalive_pool` property, I'm not familiar with this 😅 need @tzssangglass to have a confirmation.

Also, please update your first comment in English so more developers would understand your issue 😄 Hope this one will be resolved sooner or later.

**Comment by tzssangglass** [comment]
I need to be sure: does this problem still exist if the request does not go through APISIX?

**Comment by tzssangglass** [comment]
here is my step:

```
curl --location --request PUT 'http://127.0.0.1:9080/apisix/admin/routes/1' \
--header 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' \
--header 'Content-Type: application/json' \
--data-raw '{
    "uri": "/get",
    "name": "get",
    "methods": [
        "GET",
        "POST",
        "PUT",
        "DELETE",
        "PATCH",
        "HEAD",
        "OPTIONS",
        "CONNECT",
        "TRACE"
    ],
    "plugins": {
        "proxy-rewrite": {
            "regex_uri": [
                "/(.*)",
                "/get?foo1=bar1&foo2=$1"
            ]
        }
    },
    "upstream": {
        "nodes": [
            {
                "host": "httpbin.org",
                "port": 80,
                "weight": 1
            }
        ],
        "timeout": {
            "connect": 6,
            "send": 6,
            "read": 6
        },
        "type": "roundrobin",
        "scheme": "http",
        "pass_host": "node",
        "keepalive_pool": {
            "idle_timeout": 60,
            "requests": 1000,
            "size": 320
        }
    },
    "status": 1
}'
```

and use 

```
curl 127.0.0.1:9080/get
```

to test, I did not feel delay.

**Comment by tokers** [comment]
> # route config
> ```json
> {
>   "uri": "/testslow",
>   "name": "testslow",
>   "methods": [
>     "GET",
>     "POST",
>     "PUT",
>     "DELETE",
>     "PATCH",
>     "HEAD",
>     "OPTIONS",
>     "CONNECT",
>     "TRACE"
>   ],
>   "plugins": {
>     "proxy-rewrite": {
>       "regex_uri": [
>         "/(.*)",
>         "/get?foo1=bar1&foo2=$1"
>       ]
>     }
>   },
>   "upstream": {
>     "nodes": [
>       {
>         "host": "httpbin.org",
>         "port": 80,
>         "weight": 1
>       }
>     ],
>     "timeout": {
>       "connect": 6,
>       "send": 6,
>       "read": 6
>     },
>     "type": "roundrobin",
>     "scheme": "http",
>     "pass_host": "node",
>     "keepalive_pool": {
>       "idle_timeout": 60,
>       "requests": 1000,
>       "size": 320
>     }
>   },
>   "status": 1
> }
> ```
> 
> # tcpdump when use firefox
> ![image](https://user-images.githubusercontent.com/15232738/142602469-85efb3d0-1476-4369-ae03-e8ac76067f16.png)
> 
> # tcpdump when use curl
> ![image](https://user-images.githubusercontent.com/15232738/142602759-8f75d5be-a1f0-4a39-876d-5d38cf2c31cf.png)
> 
> ![image](https://user-images.githubusercontent.com/15232738/142602860-26f366a3-fd62-4c35-89f2-a438ca168ec3.png)
> 
> # It looks like a connection problem

1. Do you maintain the backend node `httpbin.org` privately, or just use the online one? If this is the public site, you should consider the network issues.
2. It seems the subsequent requests happened without the front TCP 3-way handshakes, making sure whether the slow request you said suffered from the TCP 3-way handshake, and measuring the cost time.

**Comment by aoozoo** [comment]
> 

![image](https://user-images.githubusercontent.com/15232738/142801034-54dc7473-6e4d-47f0-ae93-be5441a7fbf8.png)

Relative: 7.313300 apisix has accepted the request "Get /testslow"

What happened between 7.313300 ~ 9.992424  ？

![image](https://user-images.githubusercontent.com/15232738/142801388-4c7e10d1-7f83-42fa-8ead-9540a49fd751.png)

Start request httpbin.org at 9.514182 and finished at 9.992300

What happened between 7.313300 ~ 9.514182 ？ I don’t know how to troubleshoot this problem anymore.

Today I found that my host has opened the VPN service and I cannot reproduce this issue in my Alibaba Cloud environment.

So I turned off the vpn service of my host, and then I found that I could not reproduce the problem

It should be a problem with my test environment .

Apisix does not have this bug.

Thank you all ：）
